### PR TITLE
docs(api): added info about webpack_exports_info

### LINKED ISSUE #3267: Document webpack change: add `__webpack_exports_info__` API to allow export introspection
*A pull request by @sokra was merged and maintainers requested a documentation change.*

See pull request: https://github.com/webpack/webpack/pull/9620

---

improve usage/provide information for nested exports

improve tests for deep exports

**What kind of change does this PR introduce?**
feature

**Did you add tests for your changes?**
yes

**Does this PR introduce a breaking change?**
no

**What needs to be documented once your changes are merged?**
In modules `__webpack_exports_info__` is available:
* `__webpack_exports_info__` is always true
* `__webpack_exports_info__.<exportName>.used` is `false` when the export is known to be unused, true otherwise
* `__webpack_exports_info__.<exportName>.useInfo` is
  * `false` when the export is known to be unused
  * `true` when the export is known to be used
  * `null` when the export usage could depend on runtime conditions
  * `undefined` when no info is available
* `__webpack_exports_info__.<exportName>.provideInfo` is
  * `false` when the export is known to be not provided
  * `true` when the export is known to be provided
  * `null` when the export provision could depend on runtime conditions
  * `undefined` when no info is available
* Accessing the info from nested exports is possible: i. e. `__webpack_exports_info__.<exportName>.<exportName>.<exportName>.used`

### PR DESCRIPTION
**Pull request by anikethsaha**
closes #3267

### CODE REVIEW

File: src/content/api/module-variables.md [hunk 1]
```diff
@@ -147,6 +149,30 @@ This variable is only available with the `HotModuleReplacementPlugin` or the `Ex
 Generates a `require` function that is not parsed by webpack. Can be used to do cool stuff with a global require function if available.
 
 
+### `__webpack_exports_info__` (webpack-specific)
+
+In modules, `__webpack_exports_info__` is available
+
+- `__webpack_exports_info__` always `true`
```
Comment (jeffin143) [inline src/content/api/module-variables.md]:
```suggestion
- `__webpack_exports_info__` is always `true`
```

File: src/content/api/module-variables.md [hunk 2]
```diff
@@ -147,6 +149,30 @@ This variable is only available with the `HotModuleReplacementPlugin` or the `Ex
...
+
+In modules, `__webpack_exports_info__` is available
+
+- `__webpack_exports_info__` is always `true`
+
+- `__webpack_exports_info__.<exportName>.used` is false when the export is known to be unused, true otherwise
+
+- `__webpack_exports_info__.<exportName>.useInfo` is
+
+    - false when the export is known to be unused
```
Comment (montogeek) [inline src/content/api/module-variables.md]: Could you please add these values in backtikcs? 
`true`, `false`, etc.

File: src/content/api/module-variables.md [hunk 3]
```diff
@@ -147,6 +149,30 @@ This variable is only available with the `HotModuleReplacementPlugin` or the `Ex
 Generates a `require` function that is not parsed by webpack. Can be used to do cool stuff with a global require function if available.
 
 
+### `__webpack_exports_info__` (webpack-specific)
+
+In modules, `__webpack_exports_info__` is available
```
Comment (montogeek) [inline src/content/api/module-variables.md]:
```suggestion
In modules, `__webpack_exports_info__` is available to allow exports introspection:
```

**Comment by anikethsaha** [comment]
cc @montogeek !

**Comment by montogeek** [comment]
Thanks!

**Pull request merged**